src/test/java/calculator/NumberExtractor.java

```diff
@@ -0,0 +1,48 @@
+package calculator;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+public class NumberExtractor {
+
+    private static final String DEFAULT_DELIMITER = "[,:]";
+
+    private static final String CUSTOM_DELIMITER_PREFIX = "//";
+    private static final String CUSTOM_DELIMITER_SUFFIX = "\n";
+
+    public String extractDelimiter(final String text) {
+        if (text.startsWith(CUSTOM_DELIMITER_PREFIX)) {
+            return text.split(CUSTOM_DELIMITER_SUFFIX)[0].substring(2);
+        }
+
+        return DEFAULT_DELIMITER;
+    }
+
+    public String extractNumberText(final String text) {
+        if (text.startsWith("//")) {
+            return text.split("\n")[1];
+        }
+
+        return text;
+    }
+
+    public void validateNumber(final Integer number) {
+        if (number < 0) {
+            throw new RuntimeException("Negative numbers are not allowed");
+        }
+    }
+
+    public Numbers extract(final String text) {
+        if (text == null || text.isEmpty()) {
+            return new Numbers(List.of());
```

Comment:
비어있는 List라면 `Collections.emptyList()`를 활용해볼 수 있을 것 같아요. 🙂
```suggestion
            return new Numbers(Collections.emptyList());
```